### PR TITLE
data shield document updates

### PR DESCRIPTION
**Pull request by nithyaramakrishnan**
@smguilia @ChristinaMak - the following updates are made:

- Added content for "**Using Data Shield with OpenShift**" in the "Install" and "deploy files".
- Added sections for "**Create application**" and "**Create Build**" using API.
- Added `-Xdump:none` to the options for OpenJ9/Liberty in the Java settings section of Data Shield.
- Added a line for "Requesting Application Certificate" using Data Shield UI which was missing earlier.
- API related updates: replaced the `POST /tools/converter/convert-app` and `POST /builds` APIs with ` POST /apps` and `POST /builds/convert-app`.

### CODE REVIEW

File: convert.md [hunk 1]
```diff
@@ -190,48 +189,78 @@ To use the `Java-Mode` conversion, modify your Docker file to supply the followi
...
  ```
- {: codeblock}
+{: codeblock}
+
+## Create a build
+{: #convert-build}
+
+Run the following command to create a build of an application:
+```
+curl -H 'Content-Type: application/json' -d '{"app_id": "53fca221-b65f-4deb-ae39-20534a717b06", "docker_version": "1.15.2" }'  -H "Authorization: Bearer $em_token" https://enclave-manager.<ingress-domain>/api/v1/builds/convert-app
```
Comment (ChristinaMak) [inline convert.md]: Replace the specific example `app_id` value with <app_id> so it's easier for the user to notice to put in their value.
For `docker_version`, using <image tag> instead would be more clear.

File: convert.md [hunk 2]
```diff
@@ -190,48 +189,78 @@ To use the `Java-Mode` conversion, modify your Docker file to supply the followi
...
+                 "issuer": "MANAGER_CA",
+                 "keyType": "RSA",
+                 "keyParam": {
+                   "size": 2048
+                 },
+                 "subject": "SGX-Application.domain",
+                 "keyPath": "/appkey.pem",
+                 "certPath": "/appcert.pem",
+                 "caCertPath": "/cacert.pem",
+                 "chainPath": "/chainpath.pem"
```
Comment (ChristinaMak) [inline convert.md]: Creating a build will fail if both caCertPath and chainPath are specified, so add a warning only specify as most one of these fields.

File: convert.md [hunk 3]
```diff
@@ -190,48 +189,78 @@ To use the `Java-Mode` conversion, modify your Docker file to supply the followi
...
   ```
   {: codeblock}
 
 
-## Requesting an application certificate
-{: #request-cert}
+## Create an application and Request an application certificate
+{: #create-app-request-cert}
+
+1. once you have the `$token` you need to get the Bearer token:
```
[the remark this answers is not in the document]
Comment (nithyaramakrishnan) [reply convert.md]: @ChristinaMak , please go ahead and replace "once you have the `$token` you need to get the Bearer token:" with "Obtain an Enclave Manager access token by using the IAM authentication token:"

File: convert.md [hunk 4]
```diff
@@ -190,48 +189,78 @@ To use the `Java-Mode` conversion, modify your Docker file to supply the followi
...
-           "chainPath": "none"
-         }
-       ]
+     "name": "name of your application",
+     "description": "description of your application",
+     "input_image_name": "your-registry-server/your-app",
+     "output_image_name": "your-registry-server/your-app-sgx",
+     "isvprodid": 1,
+     "isvsvn": 1,
+     "memsize": 1024,
```
[the remark this answers is not in the document]
Comment (nithyaramakrishnan) [reply convert.md]: @ChristinaMak , it is better to provide a description with example for the values. Please see updated json below:
` {
     "name": "name of your application",
     "description": "description of your application",
     "input_image_name": "your-registry-server/your-app",
     "output_image_name": "your-registry-server/your-app-sgx",
     "isvprodid": "A numeric product identifier, choose a unique value in the range 0-65535 (Ex: 1)"
     "isvsvn": "A numeric security version to be assigned to the Enclave. This number should be incremented if security relevant change is made to the application (Ex: 1)",
     "memsize": "Memory size of application (Ex: 2048)",
     "threads": "Thread count to support the application (Ex: 128)",
     "allowed_domains": ["SGX-Application.domain"],
     "advanced_settings": {
         "encryptedDirs": ["/tmp", "/etc/app/app-dir"],
         "java_runtime": "OPENJDK",
         "certificate": {
                 "issuer": "MANAGER_CA",
                 "keyType": "RSA",
                 "keyParam": {
                   "size": 2048
                 },
                 "subject": "SGX-Application.domain",
                 "keyPath": "/appkey.pem",
                 "certPath": "/appcert.pem",
                 "caCertPath": "/cacert.pem",
                 "chainPath": "/chainpath.pem"
     }
   }
 }`

File: convert.md (hunk 1 above)
Comment (nithyaramakrishnan) [reply convert.md]: `curl -H 'Content-Type: application/json' -d '{"app_id": <app_id>, "docker_version": <docker_image_tag>" }'  -H "Authorization: Bearer $em_token" https://enclave-manager.<ingress-domain>/api/v1/builds/convert-app`

File: convert.md (hunk 2 above)
Comment (nithyaramakrishnan) [reply convert.md]: @ChristinaMak - You are right, please go ahead and add the warning.

**Comment by nithyaramakrishnan** [comment]
@ChristinaMak , once you complete the updates, can you please merge the branch so that I can get the latest dump.

File: convert.md [hunk 5]
```diff
@@ -190,48 +189,78 @@ To use the `Java-Mode` conversion, modify your Docker file to supply the followi
...
+     "threads": 64,
```
[the remark this answers is not in the document]
Comment (nithyaramakrishnan) [reply convert.md]: See updated json.

File: convert.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (smguilia) [reply convert.md]: @ChristinaMak can the memory size be anything? Or does it have to be 2048? Or is it less than that?

[the remark this answers is not in the document]
Comment (ChristinaMak) [reply convert.md]: The memory size must be a power of 2.

[the remark this answers is not in the document]
Comment (mkrause) [reply convert.md]: The property name here is wrong. Should be `mem_size`, not `memsize`.

[the remark this answers is not in the document]
Comment (smguilia) [reply convert.md]: @mkrause Fixed! Should be built in production in about 10 minutes. Thanks for pointing it out!

**Comment by smguilia** [comment]
Changes were made internally. Closing.

**Pull request closed without merging**